File: src/Frontend/web-app/online-poll-web/src/models/Poll.ts
export interface Poll {
    id: number;
    title: string;
    description: string;
    createdAt: string;
    startDate: string;
    endDate: string;
    isActive: boolean;
    creatorId: number;
    options: PollOption[];
}

export interface PollOption {
    id: number;
    pollId: number;
    optionText: string;
    voteCount: number;
}

export interface PollResult {
    pollId: number;
    title: string;
    optionResults: PollOptionResult[];
}

export interface PollOptionResult {
    optionId: number;
    optionText: string;
    voteCount: number;
    percentage: number;
}

export interface PaginationResult<T> {
    items: T[];
    totalItems: number;
    pageNumber: number;
    pageSize: number;
    totalPages: number;
}

File: src/Mobile/PollApp.Mobile/Models/Poll.cs
public class Poll
{
    public int Id { get; set; }
    public string Title { get; set; }
    public string Description { get; set; }
    public DateTime StartDate { get; set; }
    public DateTime EndDate { get; set; }
    public bool IsActive { get; set; }
    public List<PollOption> Options { get; set; }
}